app/src/playstore/java/openfoodfacts/github/scrachx/openfood/camera/GraphicOverlay.kt

```diff
@@ -0,0 +1,109 @@
+
+package openfoodfacts.github.scrachx.openfood.camera
+
+import android.content.Context
+import android.graphics.Canvas
+import android.graphics.Rect
+import android.graphics.RectF
+import android.util.AttributeSet
+import android.view.View
+import openfoodfacts.github.scrachx.openfood.utils.CameraUtils.isPortraitMode
+import java.util.ArrayList
+
+/**
+ * A view which renders custom graphics overlaid on top of an associated preview
+ * (i.e., the camera preview).
+ *
+ * Supports scaling and mirroring of the graphics relative the camera's preview properties. The
+ * idea is that detection items are expressed in terms of a preview size, but need to be scaled up
+ * to the full view size, and also mirrored in the case of the front-facing camera.
+ *
+ *
+ * Associated [Graphic] items should use [.translateX] and [ ][.translateY] to convert to view coordinate from the preview's coordinate.
+ */
+class GraphicOverlay(context: Context, attrs: AttributeSet) : View(context, attrs) {
+    private val lock = Any()
```

Comment:
Why don't we use [Locks](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/locks/Lock.html)?

Also [locks in kotlin](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.concurrent/java.util.concurrent.locks.-lock/).